internal/server/server.go

```diff
@@ -55,15 +58,14 @@ type Options struct {
 	GoogleClientID     string
 	GoogleClientSecret string
 
-	DBEncryptionKey         string
-	DBEncryptionKeyProvider string
```

Comment:
To catch any potential problems with upgrades of self-hosted infra servers we could keep this field and error if it's set to anything other than `native`.

Reply:
I've added this in the latest commit.

Reply:
Custom DB encryption key providers will be extremely rare. I think we're okay if we document dropping support for this in the release notes.

Reply:
(Current approach is good too)